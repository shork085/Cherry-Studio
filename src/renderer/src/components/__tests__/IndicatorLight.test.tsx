import { render } from '@testing-library/react'
import { describe, expect, it } from 'vitest'

import IndicatorLight from '../IndicatorLight'

describe('IndicatorLight', () => {
  it('should render with default props', () => {
    const { container } = render(<IndicatorLight color="red" />)
    const light = container.firstChild as HTMLElement

    expect(light).toHaveStyle({
      width: '8px',
      height: '8px',
      animation: 'pulse 2s infinite'
    })
  })

  it('should apply custom size', () => {
    const { container } = render(<IndicatorLight color="blue" size={16} />)
    const light = container.firstChild as HTMLElement

    expect(light).toHaveStyle({
      width: '16px',
      height: '16px'
    })
  })

  it('should convert green color to hex value', () => {
    const { container } = render(<IndicatorLight color="green" />)
    const light = container.firstChild as HTMLElement
    expect(light).toHaveStyle({ backgroundColor: '#22c55e' })
  })

  it('should disable shadow when specified', () => {
    const { container } = render(<IndicatorLight color="red" shadow={false} />)
    const light = container.firstChild as HTMLElement
    expect(light).toHaveStyle({ boxShadow: 'none' })
  })

  it('should disable animation when specified', () => {
    const { container } = render(<IndicatorLight color="green" animation={false} />)
    const light = container.firstChild as HTMLElement
    expect(light).toHaveStyle({ animation: 'none' })
  })

  it('should match snapshot', () => {
    const { container } = render(<IndicatorLight color="green" />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
