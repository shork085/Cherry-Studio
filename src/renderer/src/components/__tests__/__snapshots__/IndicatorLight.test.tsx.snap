// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IndicatorLight > should match snapshot 1`] = `
.c0 {
  width: 8px;
  height: 8px;
  border-radius: 50%;
  background-color: #22c55e;
  box-shadow: 0 0 6px #22c55e;
  animation: pulse 2s infinite;
}

<div
  class="c0"
  color="#22c55e"
  size="8"
/>
`;
